libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -104,17 +105,49 @@ bool ImmutableValidator::visit(MemberAccess const& _memberAccess)
 
 bool ImmutableValidator::visit(IfStatement const& _ifStatement)
 {
-	bool prevInBranch = m_inBranch;
+	auto prevInBranch = m_inBranch;
+	auto prevWrites = std::move(m_immutableStateVariablesWrittenInBranch);
```

Comment:
It seems dangerous to me to use std::move() here. You also access the moved variable a a few lines down and then move it again, yet to my understanding variables that have been moved should no longer be used...

Reply:
I thought about that at first, then [this post](https://stackoverflow.com/questions/7027523/what-can-i-do-with-a-moved-from-object) convinced me that it's safe to use `clear` on it, which would put it back into a valid state.

The motivation behind using a `std::move` here is to avoid unnecessary copying. It'll be a no-issue if we use an immutable map here.

Regardless, I'm still waiting to hear feedback on the alternative approach, since it'll lead to simpler code and better error message IMO.

Reply:
True, with the info in your link, the usage of `std::move` here seems okay.